src/assisted_test_infra/test_infra/controllers/ipxe_controller/ipxe_controller.py

```diff
@@ -0,0 +1,58 @@
+import os
+import shutil
+
+import consts
+from assisted_test_infra.test_infra import utils
+from service_client import log
+
+
+class IPXEController:
+    def __init__(
+        self,
+        name=None,
+        port=consts.DEFAULT_IPXE_SERVER_PORT,
+        ip=consts.DEFAULT_IPXE_SERVER_IP,
+    ):
+        self.name = name
+        self.ip = ip
+        self.port = port
+        self.dir = os.path.dirname(os.path.realpath(__file__))
+        self.ipxe_scripts_folder = f"{self.dir}/server/ipxe_scripts"
+
+    def remove(self):
+        log.info(f"Removing iPXE Server {self.name}")
+        utils.remove_running_container(container_name=self.name)
+        self._remove_ipxe_scripts_folder()
+
+    def start(self, api_client, infra_env_id, cluster_name):
+        log.info("Preparing iPXE server")
+        self.download_ipxe_script(api_client=api_client, infra_env_id=infra_env_id, cluster_name=cluster_name)
+        self.build_server_image()
+        self.run_ipxe_server()
+
+    def run_ipxe_server(self):
+        log.info(f"Running iPXE Server {self.name}")
+        run_flags = [
+            "-d",
+            "--network=host",
+            f"--publish {self.port}:{self.port}",
+        ]
+        utils.run_container(container_name=self.name, image=self.name, flags=run_flags)
+
+    def build_server_image(self):
+        log.info(f"Creating Image for iPXE Server {self.name}")
+        build_flags = f"--build-arg SERVER_IP={self.ip} --build-arg SERVER_PORT={self.port}"
+        utils.run_command(f"podman {consts.PODMAN_FLAGS} build {self.dir}/server -t {self.name} {build_flags}")
+
+    def download_ipxe_script(self, api_client, infra_env_id, cluster_name):
```

Comment:
If not used elsewhere this method can be set as protected/private

Reply:
Modified to private, thanks!